File: leetcode/122.best-time-to-buy-and-sell-stock-ii/ts/maxprofit.ts
const maxProfit = (prices: number[]): number => {
    if (prices.length === 0) return 0
    
    let res = 0

    for (let i = 1; i < prices.length; i++) {
        const profit = prices[i] - prices[i - 1]

        if (profit > 0) res += profit
    }

    return res
}

File: leetcode/122.best-time-to-buy-and-sell-stock-ii/cpp/maxprofit.cpp
#include <vector>

class Solution
{
public:
    int maxProfit(std::vector<int> &prices)
    {
        if (prices.empty())
            return 0;

        int res = 0;

        for (auto i = 1; i < prices.size(); i++)
        {
            int profit = prices[i] - prices[i - 1];

            if (profit > 0)
                res += profit;
        }

        return res;
    }
};
